autoload/vital/__vital__/VS/Vim/Window/Widget/MarkupContent.vim

```diff
@@ -0,0 +1,101 @@
+function! s:_vital_loaded(V) abort
+  let s:Markdown = a:V.import('VS.Vim.Syntax.Markdown')
+  let s:Window = a:V.import('VS.Vim.Window')
+  let s:FloatingWindow = a:V.import('VS.Vim.Window.FloatingWindow')
+endfunction
+
+function! s:_vital_depends() abort
+  return ['VS.Vim.Syntax.Markdown', 'VS.Vim.Window', 'VS.Vim.Window.FloatingWindow']
+endfunction
+
+"
+" is_available
+"
+function! s:is_available() abort
+  return s:FloatingWindow.is_available()
+endfunction
+
+"
+" new
+"
+function! s:new(args) abort
+  return s:MarkupContent.new(a:args)
+endfunction
+
+let s:id = 0
+
+let s:MarkupContent = {}
+
+"
+" new
+"
+" @param args.minwidth
+" @param args.maxwidth
+" @param args.minheight
+" @param args.maxheight
+"
+function! s:MarkupContent.new(args) abort
+  let s:id += 1
+
+  let l:bufnr = bufnr(printf('VS.Vim.Window.Widget.MarkupContent:%s', s:id), v:true)
+  call bufload(l:bufnr)
+  call setbufvar(l:bufnr, '&buflisted', 0)
+  call setbufvar(l:bufnr, '&modeline', 0)
+  call setbufvar(l:bufnr, '&buftype', 'nofile')
+  call setbufvar(l:bufnr, '&bufhidden', 'hide')
+  call setbufvar(l:bufnr, '&textwidth', 0)
+  return extend(deepcopy(s:MarkupContent), {
+  \   'bufnr': l:bufnr,
+  \   'window': s:FloatingWindow.new(),
+  \   'minwidth': get(a:args, 'minwidth', -1),
+  \   'maxwidth': get(a:args, 'maxwidth', -1),
+  \   'minheight': get(a:args, 'minheight', -1),
+  \   'maxheight': get(a:args, 'maxheight', -1),
+  \ })
+endfunction
+
+"
+" open
+"
+" @param {number} row
+" @param {number} col
+" @param {string[]} contents
+"
+function! s:MarkupContent.open(row, col, contents) abort
+  call deletebufline(self.bufnr, '^', '$')
```

Comment:
not required now but neovim has an api to do bulk edits. not sure if it is already in master or in pr.